mountpoint-s3/src/fs.rs

```diff
@@ -282,7 +282,8 @@ where
                 return Err(libc::EINVAL);
             }
 
-            lookup.inode.start_writing()?;
+            let parent_lookup = self.superblock.getattr(&self.client, lookup.inode.parent()).await?;
+            lookup.inode.start_writing(&parent_lookup.inode)?;
```

Comment:
I think we probably want to do this little "connect it to the parent" dance in `mknod`? That's when the file is "created". This `start_writing` stuff is just to decide who wins the race to start actually writing into the file.